src/sagemaker/utils.py

```diff
@@ -1066,6 +1065,15 @@ def resolve_value_from_config(
     Returns:
         The value that should be used by the caller
     """
+
+    if (
+        sagemaker_session is not None
+        and sagemaker_session.settings is not None
+        and sagemaker_session.settings.ignore_intelligent_defaults
```

Comment:
So Im wondering why we dont want to do something like the following (ignore at initialization rather than when its time to use it). Seems like this could limit potential for confusion. But curious if there's usecases you have in mind that make this approach unideal.

```
class Session(object):
...
    def _initialize(...):
       ...
        if self.settings and sagemaker_session.settings.ignore_sagemaker_config: # new
            self.sagemaker_config = {}  # new
        elif sagemaker_config:
            validate_sagemaker_config(sagemaker_config)
            self.sagemaker_config = sagemaker_config
        elif not (self.settings and self.settings.ignore_sagemaker_config):
            self.sagemaker_config = load_sagemaker_config(s3_resource=self.s3_resource)       
```